.github/workflows/ci.yml

```
@@ -0,0 +1,140 @@
+name: CI Workflow
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - rel/**
+  pull_request:
+    branches:
+      - main
+
+permissions: read-all
+
+jobs:
+  lint:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Cache pre-commit
+        uses: actions/cache@v4
+        with:
+          path: ${{ runner.home }}/.cache/pre-commit
+          key: ${{ runner.os }}-pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
+          restore-keys: ${{ runner.os }}-pre-commit-
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10.14'
+
+      - name: Install dev dependencies
+        run: pip install -r requirements-dev.txt
+
+      - name: Run pre-commit linting
+        run: pre-commit run --all-files
+
+  system-tests:
+    name: System Tests
+    runs-on: windows-latest
+    needs: [tests,lint]
+    if: github.event_name != 'pull_request' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/rel/'))
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10.8'
+
+      - name: Install test dependencies
+        run: |
+          pip install -r requirements-test.txt
+          pip install tox
+          pip install --upgrade coverage pytest pytest-cov
+
+      - name: Run system tests with tox
+        run: tox -e system_tests
+        env:
+          DAR_CLIENT_ID: ${{ secrets.DAR_CLIENT_ID }}
+          DAR_CLIENT_SECRET: ${{ secrets.DAR_CLIENT_SECRET }}
+          DAR_AUTH_URL: ${{ secrets.DAR_AUTH_URL }}
+          DAR_URL: ${{ secrets.DAR_URL }}
+
+      - name: Create Github Release for Tag
+        if: startsWith(github.ref, 'refs/tags/rel/')
+        uses: softprops/action-gh-release@v2
+        with:
+          body: "Tagging release ${{ github.ref }}"
+          # API key for a machine user specifically created for this task:
+          # https://github.com/dar-sdk-machine-user
+          token: ${{ secrets.API_KEY }}
```

```suggestion
          token: ${{ secrets.GITHUB_TOKEN }}
```

you should use here the built-in access token of GitHub Actions. Added bonus, you don't need a technical GitHub user anymore.

---

According to https://github.com/softprops/action-gh-release, we could as well remove the `token` property completly as it defaults to the GitHub token.